src/Language/DifferentialDatalog/FlatBuffer.hs

```diff
@@ -494,19 +517,38 @@ jFBType nested x =
         TOpaque{typeArgs = [elemType], ..} | elem typeName sET_TYPES
                            -> if nested
                                  then "int"
-                                 else jFBType True elemType <> "[]"
+                                 else jFBWriteType True elemType <> "[]"
         TOpaque{typeArgs = [_,_],..} | typeName == mAP_TYPE
                            -> if nested
                                  then "int"
                                  else "int []"
-        t'                 -> error $ "FlatBuffer.jConvType: unsupported type " ++ show t'
+        t'                 -> error $ "FlatBuffer.jFBWriteType: unsupported type " ++ show t'
+
+-- Java types used in the FlafBuffers-generated _de-serialization_ API
+jFBReadType :: (WithType a, ?d::DatalogProgram, ?prog_name::String) => a -> Doc
+jFBReadType x =
+    case typeNormalizeForFlatBuf x of
```

Comment:
This should be documented somewhere.

Reply:
yep, the next PR will include some documentation.